### PR TITLE
fix: negative confDepth bug in verifier

### LINKED ISSUE #123: Potential bug when confirmationDepth of -1
The [default value for confirmationDepth](https://github.com/Layr-Labs/eigenda-proxy/blob/main/server/config.go#L462) is -1. This is used to indicate to wait for finality. However, I believe that this causes a bug in getContextBlock(), where confirmationDepth is used as is, and -1 is subtracted from it before [subtracting that result from the blockNumber](https://github.com/Layr-Labs/eigenda-proxy/blob/09497a94ab764dd5a04d75d3b92ae4cb7917390a/verify/cert.go#L118).

This gives a blockNumber of `curBlockNumber+2`, which when we later query for that block number, results in a `header not found` error, and `internal server error`:
image

### PR DESCRIPTION
**Pull request by samlaf**
## Fixes Issue

Fixes #123 

## Changes proposed

Still think that this confirmation depth feature should be in the eigenda-client itself and not here (see issue #130), but this at least fixes the bug.

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -203,7 +203,15 @@ func ReadConfig(ctx *cli.Context) Config {
 		FallbackTargets:        ctx.StringSlice(FallbackTargets),
 		CacheTargets:           ctx.StringSlice(CacheTargets),
 	}
-	cfg.ClientConfig.WaitForFinalization = (cfg.EthConfirmationDepth < 0)
+	// the eigenda client can only wait for 0 confirmations or finality
+	// the da-proxy has a more fine-grained notion of confirmation depth
+	// we use -1 to let the da client wait for finality, and then need to set the confirmation depth
+	// for the da-proxy to 0 (because negative confirmation depth doesn't mean anything and leads to errors)
+	// TODO: should the eigenda-client implement this feature for us instead?
```
Comment (epociask) [inline server/config.go:210]: wouldn't that also require the client to do the cert verifications as well?

**Review by epociask — APPROVED** [review]
just one general comment - otherwise LGTM! Wonderful catch on this 🚀

File: server/config.go (hunk 1 above)
Comment (samlaf) [reply server/config.go:210]: Let's chat about this! I don't understand why you are saying this (don't have a full model of how all of these things are tied together), but I created https://github.com/Layr-Labs/eigenda-proxy/issues/130 specifically to upstream bunch of these things to the da-client. Is that possible? Is there a reason we haven't done that?

Comment (samlaf) [reply server/config.go:210]: Merging. Let's continue discussion in that issue.

**Pull request merged**